### PR TITLE
Bugfix/modify checkout process

### PR DESCRIPTION
**Pull request by HLTVProxy**
- modify CheckoutProductModel for coin merchandise purchase
- also need to watch other repos which are app, admin and backend

### CODE REVIEW

**Comment by HLTVProxy** [comment]
[(3) 【商品】商品希望也能用代幣購買 [1]](https://trello.com/c/IN2yZDyB/3475-3-%E3%80%90%E5%95%86%E5%93%81%E3%80%91%E5%95%86%E5%93%81%E5%B8%8C%E6%9C%9B%E4%B9%9F%E8%83%BD%E7%94%A8%E4%BB%A3%E5%B9%A3%E8%B3%BC%E8%B2%B7-1)

File: src/components/modals/CheckoutProductModal.tsx [hunk 1]
```diff
@@ -95,12 +95,19 @@ const CheckoutProductItem: React.VFC<{
...
+  defaultProductId?: string
+}> = ({ name, price, currencyId, quantity, saleAmount, defaultProductId }) => {
   return (
     <div className="d-flex align-items-center justify-content-between">
       <span className="flex-grow-1 mr-4">
         {name}
-        {quantity && saleAmount && <span>{` X${quantity} (${quantity * saleAmount}張)`}</span>}
+        {quantity && saleAmount && (
+          <span>{` X${quantity} ${
+            defaultProductId !== undefined && defaultProductId.includes('MerchandiseSpec_')
```
Comment (Ponchimeow) [inline src/components/modals/CheckoutProductModal.tsx:106]: NITS: can reduce it to `defaultProductId?.includes('MerchandiseSpec_') ? '' : `(${quantity * saleAmount} 張)`  `

File: src/components/modals/CheckoutProductModal.tsx [hunk 2]
```diff
@@ -615,7 +679,7 @@ const CheckoutProductModal: React.VFC<CheckoutProductModalProps> = ({
               )}
             </StyledCheckoutBlock>
             <StyledCheckoutPrice className="mb-3">
-              <PriceLabel listPrice={totalPrice} />
+              {isCoinsEnough ? <PriceLabel listPrice={totalPrice} /> : `${settings['coin.unit']} 不足`}
```
Comment (Ponchimeow) [inline src/components/modals/CheckoutProductModal.tsx:682]: if settings['coin.unit'] is not set in hasura, it will be error

**Review by Ponchimeow — CHANGES_REQUESTED** [review]
settings['coin.unit'] need to be fix

**Review by Ponchimeow — APPROVED** [review]

**Pull request merged**